beeline-core/src/test/java/io/honeycomb/beeline/builder/BeelineBuilderTest.java

```diff
@@ -326,6 +354,32 @@ public void emptyServiceNameGetsDefaultServiceName() {
         completeNegativeVerification();
     }
 
+    @Test
+    public void emptyServiceNameGetsDefaultServiceNameNonClassic() {
```

Comment:
```suggestion
    public void emptyServiceNameGetsDefaultDatasetNonClassic() {
```

Reply:
For this one I actually did intend to keep ServiceName, as it's the NonClassic version of the test above it for `emptyServiceNameGetsDefaultServiceNameClassic`.

The first line asserts that service name starts with `unknown_service`, and a couple lines down it then confirms dataset is then `unknown_service`. There may be too many things in each of these tests which makes it more difficult to see what is actually being tested in each case. I might want a follow-up to clean these up a bit to remove unnecessary test duplication.

Reply:
nah, I don't think it's worth the effort to spend more time here 😄